### ISSUE TITLE
[BUG] Use of `>=` style dependencies for aws crates breaking downstream builds

### ISSUE THREAD

**Issue by bcpeinhardt** [opensearch-project/opensearch-rs#223]
### What is the bug?
Presently, in opensearch version 2.2, aws dependencies are being specified using `>=` like so
```
aws-credential-types = { version = ">= 0.57, <1.0", optional = true }
aws-sigv4 = { version = ">= 0.57, <1.0", optional = true }
aws-smithy-runtime-api = { version = ">= 0.57, <1.0", optional = true, features = ["client"]}
aws-types = { version = ">= 0.57, <1.0", optional = true }
```
the `<1.0` was added later (presumably to fix downstream build issues being caused by bringing in v1.x.x).
Here is the commit: https://github.com/opensearch-project/opensearch-rs/commit/5a2ae455bd6d55bcc8b73a50fdf03a1c6405dbd7

I'm hoping this fix can be back ported to opensearch version 2.1. The lack of the proper pinning is breaking our downstream
builds, but upgrading to 2.2 would require upgrading aws-* dependency versions in other libraries.

**Comment by Xtansia** [comment]
Hi @bcpeinhardt, Thanks for reporting this, I'll work on backporting a fix to `2.1`, in the meantime I believe you should be able to workaround this by pinning to the correct versions in your crate, as it'll prefer to use an already requested compatible version rather than introduce a new version:
```toml
[dependencies]
opensearch = { version = "2.1.0", features = ["aws-auth"] }
aws-sigv4 = "0.55"
aws-types = "0.55"
aws-config = "0.55"
aws-credential-types = "0.55"
```

**Comment by Xtansia** [comment]
@bcpeinhardt I've now released a v2.1.1 that correctly bounds the `aws-*` dependencies to the code-compatible versions
